### PR TITLE
chore!: Update node versions to align with eslint v9

### PR DESCRIPTION
**Pull request by scagood**
**What changes did you make?**

This simply updates the CI and package.json to run the tests against the node versions that ESLint v9 supports

BREAKING CHANGE: Requires Node.js: `^18.18.0 || ^20.9.0 || >=21.1.0`

### CODE REVIEW

**Comment by brettz9** [comment]
LGTM, but maybe someone on the core team should review this kind of breaking change.

**Comment by brettz9** [comment]
@aladdin-add , @MichaelDeBoey , @voxpelli : Hi there... not sure who to ping, but I figured someone on the core team should be reviewing breaking changes?

File: package.json
Comment (voxpelli) [inline package.json:87]: Should we align here as well? Supporting all the way back to ESLint 7 but only Node 18 and newer feels a bit odd

**Review by voxpelli — COMMENTED** [review]
I’m 👍 on aligning engine range with ESLint 9

What have the other plugins in this org done? Such as plugin-n?

I have been meaning to propose a change like this for eslint-utils as well as that’s the main thing holding my PR there back https://github.com/eslint-community/eslint-utils/pull/204

**Comment by voxpelli** [comment]
> Hi there... not sure who to ping, but I figured someone on the core team should be reviewing breaking changes?

Pinging me is very okay at least :)

The governance docs got stuck a bit with people having different views and as far as I remember no counter proposals being made, so we do not have a structured process for this yet.

Feel very free to give feedback on https://github.com/eslint-community/governance/pull/1, either directly there or to me, would love to have a clear process for these kinds of things

File: package.json
Comment (scagood) [reply package.json:87]: Yeah, this is what I did in the original PR (#471)

https://github.com/eslint-community/eslint-plugin-promise/commit/4db7f5d8379aa3a04cff72145f7681e97ccdaf3d

But I think that making that a separate PR for the change log was a good idea 😀

**Review by aladdin-add — APPROVED** [review]

**Comment by aladdin-add** [comment]
leaving it open to let @xjamundx merge breaking changes, when he thinks it's a good time to release a major. :)

**Comment by voxpelli** [comment]
> leaving it open to let @xjamundx merge breaking changes, when he thinks it's a good time to release a major. :)

@aladdin-add Are you sure @xjamundx is still the one leading this module? I thought he handed it over to the ESLint Community because he wasn't interested in maintaining it anymore?

**Comment by aladdin-add** [comment]
ah, I didn't know that!  Do you think we can release a major version now? Are there any other breaking changes that should be included?

**Comment by scagood** [comment]
Node 22.5.0 has struck again! I guess I can only wait for the action to work out that 22.5.1 has been released :thinking:
(https://github.com/nodejs/node/issues/53902)

**Comment by voxpelli** [comment]
> Node 22.5.0 has struck again! I guess I can only wait for the action to work out that 22.5.1 has been released 🤔 ([nodejs/node#53902](https://github.com/nodejs/node/issues/53902))

If you tell it to use 22.5.1 specifically, then it will pass, but will have to be reverted before merging

**Comment by aladdin-add** [comment]
It's all green now! ✅

**Comment by ota-meshi** [comment]
It appears that the merge of this PR did not release successfully.
To release a major version, `BREAKING CHANGE: ` must be included in the message footer, but this repository appears to have been configured to include only the PR title in the commit message, so `BREAKING CHANGE: ` was not included.
I will revert this PR twice and try to release it.

https://github.com/semantic-release/semantic-release#commit-message-format

**Comment by aladdin-add** [comment]
it's already configured, not sure why not working as expected🤔:

image

**Comment by ota-meshi** [comment]
I just changed that setting 😅

**Comment by voxpelli** [comment]
> It appears that the merge of this PR did not release successfully. To release a major version, `BREAKING CHANGE: ` must be included in the message footer

Doing `feat!:` should be enough? The conventional commit convention says that the `!` indicates a breaking change? Does the semver release flow not support that?

_(I would personally be in favor of switching to a release please flow, like in https://github.com/eslint-community/eslint-plugin-n/pull/305, as that catches errors like this before release)_

**Pull request merged**
